### ISSUE TITLE
Difference between dropout and eval flags?!

### ISSUE THREAD

**Issue by kalai2033** [junyanz/pytorch-CycleGAN-and-pix2pix#835]
I trained a pix2pix model with --no_dropout flag. And i got same output every time when i tested the images. And when i used --eval flag while testing the model trained without --no_dropout flag, i had the same behaviour. So what is difference between --eval and --no_dropout flags

**Comment by junyanz** [comment]
`--no_dropout` specifies the existence of the Dropout layer, and `--eval` specifies the behavior of the Dropout layer (if it exists) during evaluation mode. You will get stochastic output if (1) you train the model without `--no_dropout` and test the model without `--eval`.

**Comment by vgrabovets** [comment]
`eval` also disables normalization layer. In my project I use dropout during training, but not during testing and `batch_norm` during both phases.
When I used `eval` mode, my testing results visually got worse because it disabled normalization.
